[plasma-5.2.1] make dep on qtwebkit optional in libksysguard and

ff2000 opened the pull request:
plasma-workspace

ff2000 commented:
For now I only applied the changes on 5.2.1. live ebuilds will be done if you think the patch is OK.
I then will also add USE descriptions for the newly introduced USE flags.

ff2000 commented:
Here is a diff that only contains the actual changes.
```
--- plasma-workspace-5.2.1.ebuild	2015-02-28 11:05:21.488413615 +0100
+++ plasma-workspace-5.2.1-r1.ebuild	2015-02-28 20:48:22.394942513 +0100
@@ -10,7 +10,7 @@
 
 DESCRIPTION="KDE Plasma workspace"
 KEYWORDS=" ~amd64"
-IUSE="dbus gps prison qalculate X"
+IUSE="dbus drkonqi gps prison qalculate systemmonitor X"
 
 COMMON_DEPEND="
 	$(add_plasma_dep baloo)
@@ -31,7 +31,6 @@
 	$(add_frameworks_dep kdeclarative)
 	$(add_frameworks_dep kdelibs4support)
 	$(add_frameworks_dep kdesu)
-	$(add_frameworks_dep kdewebkit)
 	$(add_frameworks_dep kglobalaccel)
 	$(add_frameworks_dep kguiaddons)
 	$(add_frameworks_dep ki18n)
@@ -62,16 +61,23 @@
 	dev-qt/qtnetwork:5
 	dev-qt/qtscript:5
 	dev-qt/qtsql:5
-	dev-qt/qtwebkit:5
 	dev-qt/qtwidgets:5
 	dev-qt/qtxml:5
 	media-libs/phonon[qt5]
 	sys-libs/pam
 	sys-libs/zlib
 	dbus? ( dev-libs/libdbusmenu-qt[qt5] )
+	drkonqi? (
+		$(add_frameworks_dep kdewebkit)
+		dev-qt/qtwebkit:5
+	)
 	gps? ( sci-geosciences/gpsd )
 	prison? ( media-libs/prison:5 )
 	qalculate? ( sci-libs/libqalculate )
+	systemmonitor? (
+		$(add_plasma_dep libksysguard processui)
+		dev-qt/qtwebkit:5
+	)
 	X? (
 		dev-qt/qtx11extras:5
 		x11-libs/libICE
@@ -117,6 +123,14 @@
 
 	# bug 539922, moved to frameworks
 	comment_add_subdirectory kglobalaccel
+
+	if ! use drkonqi; then
+		comment_add_subdirectory drkonqi
+	fi
+
+	if ! use systemmonitor; then
+		comment_add_subdirectory systemmonitor
+	fi
 }
 
 src_configure() {
--- libksysguard-5.2.1.ebuild	2015-02-28 11:05:21.487413615 +0100
+++ libksysguard-5.2.1-r1.ebuild	2015-02-28 20:48:22.373942514 +0100
@@ -11,7 +11,7 @@
 DESCRIPTION="Task management and system monitoring library"
 LICENSE="LGPL-2+"
 KEYWORDS=" ~amd64"
-IUSE="X"
+IUSE="processui X"
 
 COMMON_DEPEND="
 	$(add_frameworks_dep kauth)
@@ -26,7 +26,9 @@
 	dev-qt/qtdbus:5
 	dev-qt/qtgui:5
 	dev-qt/qtnetwork:5
-	dev-qt/qtwebkit:5
+	processui? (
+		dev-qt/qtwebkit:5
+	)
 	dev-qt/qtwidgets:5
 	sys-libs/zlib
 	X? (
@@ -45,6 +47,12 @@
 	X? ( x11-proto/xproto )
 "
 
+src_prepare() {
+	if ! use processui; then
+		comment_add_subdirectory processui
+	fi
+}
+
 src_configure() {
 	local mycmakeargs=(
 		$(cmake-utils_use_find_package X X11)
--- ksysguard-5.2.1.ebuild	2015-02-25 08:48:23.645996185 +0100
+++ ksysguard-5.2.1-r1.ebuild	2015-02-28 20:48:22.347942515 +0100
@@ -13,7 +13,7 @@
 IUSE="lm_sensors"
 
 DEPEND="
-	$(add_plasma_dep libksysguard)
+	$(add_plasma_dep libksysguard processui)
 	$(add_frameworks_dep kcompletion)
 	$(add_frameworks_dep kconfig)
 	$(add_frameworks_dep kconfigwidgets)
```

kensington commented:
This sort of change is safe to do in-place - the revision bump isn't necessary. Also, we should consider enabling these new flags by default - one would expect these features in KDE unless they deliberately turn them off.

kensington commented on kde-plasma/libksysguard/libksysguard-5.2.1-r1.ebuild:
```diff
@@ -0,0 +1,62 @@
+	if ! use processui; then
+		comment_add_subdirectory processui
+	fi
+}
```
Make sure the eclass magic can still happen:

        fi
        
        kde5_src_prepare
    }

ff2000 commented:
PR updated as requested.
I also added the changes to the live ebuilds.
I could not find a use.local.desc, so I added descriptions to the commit message.

(And I hope nothing else went wrong, because I had to squash + push as root... Used the repo for doing the changes, then something seems to be gone wrong. Need to get a better way to do the development in my $HOME but get the changes applied to the repo.)

kensington commented:
Thanks, merged with some minor style change.
